### DISCUSSION TITLE
Auditory categorization experiment with preview of options

### DISCUSSION THREAD

**Discussion by holgerMitterer** [jspsych/jsPsych#820] [Q&A] [answered]
Hi there,

I am trying to implement a simple 2AFC task in the auditory modality.
In the long run, there will be multiple options over trials, that is,
on one trial they have to decide whehter they hear "rice" or "rife",
on the next, it is about "nice" vs. "knife".

To make the task easier, I like to preview the options, but there is no
"prompt preview" parameter in the 'audio-keyboard-response' plug-in.
I therefore used a preceding "html-keyboard response'. This works,
however, only if there is actually a response, if I set a trial duration,
this generate an error. (Can only test on windows, but occured in both
Chrome and Firefox.)

That is, the following works:

 .....
    var show_options = {
        type: 'html-keyboard-response',
        stimulus: 'rice  &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; rife',
		/*choices: jsPsych.NO_KEYS,
		trial_duration: 1000*/
    }

	var twoAFC_trial = {
        type: 'audio-keyboard-response',
        stimulus: myAudioFiles[0],
		prompt: 'rice&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; rife',
		choices: ['z','m']		
    }
   timeline.push(show_options, twoAFC_trial);
....... 

but if I un-comment the two lines choices...trial duration in the show options variable,
(which is how I want the experiment to run in the end)
the sound is not played and the following error is observed:

Autoplay is only allowed when approved by the user, the site is activated by the user, or media is muted. jspsych-audio-keyboard-response.js:151:12
Uncaught (in promise) DOMException: The play method is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.

(I could upload the whole script, but it would also require the wav file ....)
I really struggle to understand what happens here, especially as a change in the first part of the timeline
impact the second.

Best regards,
Holger

**Comment by conbainbridge** [comment, marked as answer]
While I'm not able to reproduce this exact same error, I get a similar error if I do not first give the user a chance to interact with the page. For studies I've worked on we typically bypass this issue by having a consent form where they must click a button to continue. Could you try and see if adding an initial trial that requires a key press or button press fixes this?

**Reply by holgerMitterer** [reply]
Yes it does, thank you, would never have thought of that.
In the final experiment there will obviously be an instruction requiring a button press, so that is no problem.
(I simply started bottom-up, not that experience that I can program top-down very well)

**Reply by jodeleeuw** [reply]
Just to elaborate a bit, the reason for this issue is that some browsers elected to disallow audio playback until the user interacts with a page. 

Here's Chrome's statement about this: https://developers.google.com/web/updates/2017/09/autoplay-policy-changes
